dojo/importers/base_importer.py

```diff
@@ -0,0 +1,926 @@
+        reactivated_finding_count: int = 0,
+        untouched_finding_count: int = 0,
+        **kwargs: dict,
+    ) -> str:
+        """
+        Constructs a success message to be displayed on screen in the UI as a digest for the user.
+        This digest includes counts for the findings in the following status:
+        - Created: New findings that have not been created before
+        - Closed: Findings that were not detected in the report any longer, so the original was closed
+        - Reactivated: Findings that were once closed, but has reappeared in the report again
+        - Untouched: Findings that have not changed between now, and the last import/reimport
+        """
+        # Only construct this message if there is any change in finding status
+        if finding_count > 0:
+            # Set the base message to indicate how many findings were parsed from the report
+            message = f"{scan_type} processed a total of {finding_count} findings"
+            if import_type == Test_Import.IMPORT_TYPE:
+                # Check for close old findings context to determine if more detail should be added
+                if kwargs.get("close_old_findings", False):
+                    message += f" and closed {closed_finding_count} findings"
+            if import_type == Test_Import.REIMPORT_TYPE:
+                # Add more details for any status changes recorded
+                if new_finding_count:
+                    message += f" created {new_finding_count} findings"
+                if closed_finding_count:
+                    message += f" closed {closed_finding_count} findings"
+                if reactivated_finding_count:
+                    message += f" reactivated {reactivated_finding_count} findings"
+                if untouched_finding_count:
+                    message += f" did not touch {untouched_finding_count} findings"
+            # Drop a period at the end
+            message += "."
+        else:
+            # Set the message to convey that all findings processed are identical to the last time an import/reimport occurred
+            message = "No findings were added/updated/closed/reactivated as the findings in Defect Dojo are identical to those in the uploaded report."
+
+        return message
+
+    def chunk_objects(
+        self,
+        object_list: List[Finding | Endpoint],
+        chunk_size: int = settings.ASYNC_FINDING_IMPORT_CHUNK_SIZE,
+    ) -> List[List[Finding | Endpoint]]:
+        """
+        Split a single large list into a list of lists of size `chunk_size`.
+        For Example
+        ```
+        >>> chunk_objects([A, B, C, D, E], 2)
+        >>> [[A, B], [B, C], [E]]
+        ```
+        """
+        # Break the list of parsed findings into "chunk_size" lists
+        chunk_list = [list[i:i + chunk_size] for i in range(0, len(list), chunk_size)]
+        logger.debug('IMPORT_SCAN: Split endpoints into ' + str(len(chunk_list)) + ' chunks of ' + str(chunk_size))
```

Comment:
Looks like `list` is a typo here. Also, since this list could contain either Findings or Endpoints, maybe something like this in the log message? There might be a cleaner way to do this...

```suggestion
        if len(object_list) == 0:
            return object_list
        # Break the list of parsed findings into "chunk_size" lists
        chunk_list = [object_list[i:i + chunk_size] for i in range(0, len(object_list), chunk_size)]
        object_type = type(object_list[0]).__name__
        logger.debug(f'IMPORT_SCAN: Split {object_type}s into ' + str(len(chunk_list)) + ' chunks of ' + str(chunk_size))
```

Reply:
Fixed in an upcoming commit